var/spack/repos/builtin/packages/zfp/package.py

```diff
@@ -6,26 +6,45 @@
 from spack import *
 
 
-class Zfp(CMakePackage):
-    """zfp is an open source C/C++ library for high-fidelity, high-throughput
-       lossy compression of floating-point and integer multi-dimensional
-       arrays.
+class Zfp(CMakePackage, CudaPackage):
+    """zfp is a compressed format for representing multidimensional
+       floating-point and integer arrays. zfp provides compressed-array
+       classes that support high throughput read and write random access
+       to individual array elements. zfp also supports serial and parallel
+       (OpenMP and CUDA) compression of whole arrays.
     """
 
-    homepage = 'http://computing.llnl.gov/projects/floating-point-compression'
-    url      = 'http://computing.llnl.gov/projects/floating-point-compression/download/zfp-0.5.2.tar.gz'
-    git = "https://github.com/LLNL/zfp.git"
+    # Package info
+    homepage    = 'https://zfp.llnl.gov'
+    url         = 'https://github.com/LLNL/zfp/releases/download/0.5.5/zfp-0.5.5.tar.gz'
+    git         = 'https://github.com/LLNL/zfp.git'
+    maintainers = ['lindstro', 'GarrettDMorrison']
 
-    version('develop', branch='develop')
+    # Versions
     version('0.5.5', sha256='fdf7b948bab1f4e5dccfe2c2048fd98c24e417ad8fb8a51ed3463d04147393c5')
-    version('0.5.4', sha256='768a05ed9bf10e54ac306f90b81dd17b0e7b13782f01823d7da4394fd2da8adb')
-    version('0.5.2', sha256='89e718edb966422b5898b5c37b1b0a781d4effacb511520558469e3ff7f65d7c')
-    version('0.5.1', sha256='867c04cf965f1c70d9725b396c6e1b5d29db55b0d69b8e87a995aaebd221b830')
-
-    variant('bsws',
-            default='64',
-            values=('8', '16', '32', '64'),
-            multi=False,
+    version('0.5.4', sha256='746e17aaa401c67dcffd273d6e6f95c76adfbbd5cf523dcad56d09e9d3b71196')
+    version('0.5.3', sha256='a5d2f8e5b47a7c92e2a5775b82cbfb3a76c87d0ac83d25abb4ac10ea75a2856e')
+    version('0.5.2', sha256='9c738ec525cc76b4bb80b2b3f7c9f07507eeda3a341470e5942cda97efbe9a4f', url='https://github.com/LLNL/zfp/archive/0.5.2/zfp-0.5.2.tar.gz')
+    version('0.5.1', sha256='f255dd1708c9ae4dc6a56dd2614e8b47a10d833c87fd349cbd47545a19c2b779', url='https://github.com/LLNL/zfp/archive/0.5.1/zfp-0.5.1.tar.gz')
+    version('develop', branch='develop')
+
+    # Build targets
+    variant('utilities', default=True,  description='Build utilities')
+    variant('examples',  default=False, description='Build examples')
+    variant('tests',     default=True,  description='Build tests')
+    variant('shared',    default=True,  description='Build shared libraries')
+
+    # Language bindings
+    variant('c',       default=False, description='Enable C bindings')
+    variant('python',  default=False, description='Enable Python bindings')
+    variant('fortran', default=False, description='Enable Fortran bindings')
```

Comment:
Just a comment, not a request. If it makes sense for this package these three Boolean variants can be also written as a single multi-valued variant:
```suggestion
    variant(
        'bindings', values=any_combination_of('c', 'python', 'fortran'),
        description='Language bindings to enable'
    )
```

Reply:
I think that would be fine.  In the future, we may add bindings for different components of zfp, in which case we'd have to be more specific about what each variant enables.  But I guess that could be handled similarly.  I'm fine with making this change.

Reply:
In the end, I decided against this.  We have similar Boolean variants for build targets, backends, internal configuration parameters, etc.  Some combinations of these are not as straightforward to categorize.